cmd/uplink/cmd/setup.go

```diff
@@ -122,49 +154,58 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 
 // ApplyDefaultHostAndPortToAddrFlag applies the default host and/or port if either is missing in the specified flag name.
 func ApplyDefaultHostAndPortToAddrFlag(cmd *cobra.Command, flagName string) error {
-	defaultHost, defaultPort, err := net.SplitHostPort(cmd.Flags().Lookup(flagName).DefValue)
+	flag := cmd.Flags().Lookup(flagName)
+	if flag == nil {
+		// No flag found for us to handle.
+		return nil
+	}
+
+	address, err := ApplyDefaultHostAndPortToAddr(flag.Value.String(), flag.DefValue)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	flag := cmd.Flags().Lookup(flagName)
-	if flag == nil {
-		// No flag found for us to handle.
+	if flag.Value.String() == address {
+		// Don't trip the flag set bit
 		return nil
 	}
-	address := flag.Value.String()
+
+	return Error.Wrap(flag.Value.Set(address))
+}
+
+// ApplyDefaultHostAndPortToAddr applies the default host and/or port if either is missing in the specified address.
+func ApplyDefaultHostAndPortToAddr(address, defaultAddress string) (string, error) {
+	defaultHost, defaultPort, err := net.SplitHostPort(defaultAddress)
+	if err != nil {
+		return "", Error.Wrap(err)
+	}
 
 	addressParts := strings.Split(address, ":")
```

Comment:
let's use here `net.SplitHostPort` here as well

Reply:
We can't do that. See the [details here](https://github.com/storj/storj/pull/1723) why I had to do it this way.

Reply:
one problem worth considering still egon brings up is if `address` is an ipv6 address, this code won't work correctly, which net.SplitHostPort attempts to fix

Reply:
Ahh right. Depends what we want to do then.`net.SplitHostPort()` won't work with `:8888`. If we want to use `net.SplitHostPort()` we lose that ability.

If we want to retain that use case we could check for multiple `:` instances. It's a bit ugly but it would work.

Reply:
i think we'll probably always have the hostname, so if we had to pick between supporting ipv6 and supporting the no-hostname case, i'd probably pick the latter. but okay to leave for another pr

Reply:
PS: when unable to do the obvious thing leave comment for why it cannot be used, helps to avoid someone accidentally "fixing it".